esphome/components/tuya/climate/__init__.py

```diff
@@ -99,27 +157,40 @@ def validate_eco_values(value):
             cv.Optional(CONF_SUPPORTS_HEAT, default=True): cv.boolean,
             cv.Optional(CONF_SUPPORTS_COOL, default=False): cv.boolean,
             cv.Optional(CONF_SWITCH_DATAPOINT): cv.uint8_t,
-            cv.Optional(CONF_ACTIVE_STATE_DATAPOINT): cv.uint8_t,
-            cv.Optional(CONF_ACTIVE_STATE_HEATING_VALUE, default=1): cv.uint8_t,
-            cv.Optional(CONF_ACTIVE_STATE_COOLING_VALUE): cv.uint8_t,
+            cv.Optional(CONF_ACTIVE_STATE): ACTIVE_STATES,
             cv.Optional(CONF_HEATING_STATE_PIN): pins.gpio_input_pin_schema,
             cv.Optional(CONF_COOLING_STATE_PIN): pins.gpio_input_pin_schema,
             cv.Optional(CONF_TARGET_TEMPERATURE_DATAPOINT): cv.uint8_t,
             cv.Optional(CONF_CURRENT_TEMPERATURE_DATAPOINT): cv.uint8_t,
             cv.Optional(CONF_TEMPERATURE_MULTIPLIER): cv.positive_float,
             cv.Optional(CONF_CURRENT_TEMPERATURE_MULTIPLIER): cv.positive_float,
             cv.Optional(CONF_TARGET_TEMPERATURE_MULTIPLIER): cv.positive_float,
-            cv.Optional(CONF_ECO_DATAPOINT): cv.uint8_t,
-            cv.Optional(CONF_ECO_TEMPERATURE): cv.temperature,
             cv.Optional(CONF_REPORTS_FAHRENHEIT, default=False): cv.boolean,
+            cv.Optional(CONF_PRESET): PRESETS,
+            cv.Optional(CONF_FAN_MODE): FAN_MODES,
+            cv.Optional(CONF_SWING_MODE): SWING_MODES,
+            cv.Optional("active_state_datapoint"): cv.invalid(
+                "'active_state_datapoint' has been moved inside of the 'active_state' config block as 'datapoint'"
+            ),
+            cv.Optional("heating_state_datapoint"): cv.invalid(
```

Comment:
This is not a datapoint (also for cooling, and the message itself
```suggestion
            cv.Optional("active_state_heating_value"): cv.invalid(
```

Reply:
Thanks for the oversight. Missed that one.